C error when casting index and returning a pointer on a fixed length array

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: macos, macOS, 11.6, 20G165
Processor: 16 cpus, 64bit, little endian, Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
CC version: Apple clang version 13.0.0 (clang-1300.0.29.3)

getwd: /Users/wes/dev/test-v
vmodules: /Users/wes/.vmodules
vroot: /Users/wes/dev/v
vexe: /Users/wes/dev/v/v
vexe mtime: 2022-01-10 16:03:41
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 65eb383

Git version: git version 2.30.1 (Apple Git-130)
Git vroot status: weekly.2022.02-2-g65eb383c
.git/config present: true
thirdparty/tcc status: thirdparty-macos-amd64 93fa86be

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg main.v`
```v
type Entity = int

struct Component {
	vals [256]int
}

fn (c Component) get_broken(e Entity) &int {
	return &c.vals[int(e)]
}
```

**What did you expect to see?**

successful compilation


**What did you see instead?**
```
==================
   ^~~~~~~~~~~~~~~
/tmp/v_501/main.14493015409709803414.tmp.c:10882:48: error: too few arguments to function call, expected 2, have 1
        int* _t1 = &(*(c)).vals[v_fixed_index((int)(e)), 256)];
                                ~~~~~~~~~~~~~         ^
/tmp/v_501/main.14493015409709803414.tmp.c:6281:28: note: 'v_fixed_index' declared here
inline VV_LOCAL_SYMBOL int v_fixed_index(int i, int len) {
                           ^
/tmp/v_501/main.14493015409709803414.tmp.c:10882:54: error: expected ']'
        int* _t1 = &(*(c)).vals[v_fixed_index((int)(e)), 256)];
                                                            ^
/tmp/v_501/main.14493015409709803414.tmp.c:10882:25: note: to match this '['
        int* _t1 = &(*(c)).vals[v_fixed_index((int)(e)), 256)];
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -4437,10 +4437,6 @@ fn (mut g Gen) ident(node ast.Ident) {
 }
 
 fn (mut g Gen) cast_expr(node ast.CastExpr) {
-	if node.typname.starts_with('&') {
-		// &Foo(0) => ((Foo*)0)
-		g.out.go_back(1)
-	}
 	sym := g.table.sym(node.typ)
 	if sym.kind in [.sum_type, .interface_] {
 		g.expr_with_cast(node.expr, node.expr_type, node.typ)

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -4437,11 +4437,10 @@ fn (mut g Gen) ident(node ast.Ident) {
 }
 
 fn (mut g Gen) cast_expr(node ast.CastExpr) {
-	if g.is_amp {
+	if node.typname.starts_with('&') {
 		// &Foo(0) => ((Foo*)0)
 		g.out.go_back(1)
 	}
-	g.is_amp = false
 	sym := g.table.sym(node.typ)
 	if sym.kind in [.sum_type, .interface_] {
 		g.expr_with_cast(node.expr, node.expr_type, node.typ)

--- a/vlib/v/tests/cast_in_index_of_ref_fixed_array_test.v
+++ b/vlib/v/tests/cast_in_index_of_ref_fixed_array_test.v
@@ -1,0 +1,16 @@
+type Entity = int
+
+struct Component {
+	vals [256]int
+}
+
+fn (c Component) get_broken(e Entity) &int {
+	return &c.vals[int(e)]
+}
+
+fn test_cast_in_ref_fixed_array() {
+	c := Component{}
+	a := c.get_broken(Entity(10))
+	println(*a)
+	assert *a == 0
+}

